android/src/com/unciv/app/FontAndroid.kt

```diff
@@ -0,0 +1,163 @@
+package com.unciv.app
+
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.Typeface
+import android.graphics.fonts.Font
+import android.graphics.fonts.FontFamily
+import android.graphics.fonts.FontStyle
+import android.graphics.fonts.SystemFonts
+import android.os.Build
+import android.util.Log
+import androidx.annotation.RequiresApi
+import com.badlogic.gdx.Gdx
+import com.badlogic.gdx.graphics.Pixmap
+import com.unciv.ui.components.FontFamilyData
+import com.unciv.ui.components.FontImplementation
+import com.unciv.ui.components.Fonts
+import java.util.*
+import kotlin.math.abs
+
+class FontAndroid : FontImplementation {
+
+    private val fontList: HashSet<Font> = hashSetOf()
+    private val paint: Paint = Paint()
+    private var currentFontFamily: String? = null
+
+    init {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q)
+            fontList.addAll(SystemFonts.getAvailableFonts())
+        paint.isAntiAlias = true
+        paint.strokeWidth = 0f
+        paint.setARGB(255, 255, 255, 255)
+
+    }
+
+    override fun setFontFamily(fontFamilyData: FontFamilyData, size: Int) {
+        paint.textSize = size.toFloat()
+
+        // Don't have to reload typeface if font-family didn't change
+        if (currentFontFamily != fontFamilyData.invariantName) {
+            currentFontFamily = fontFamilyData.invariantName
+
+            // Mod font
+            if (fontFamilyData.filePath != null)
+            {
+                paint.typeface = createFontFromFile(fontFamilyData.filePath!!)
+            }
+            // System font
+            else
+            {
+                val fontName = fontFamilyData.invariantName
+                paint.typeface =
+                        if (fontName.isNotBlank() && Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q)
+                        {
+                            val font = fontList.mapNotNull {
+                                val distanceToRegular = it.matchesFamily(fontName)
+                                if (distanceToRegular == Int.MAX_VALUE) null else it to distanceToRegular
+                            }.minByOrNull { it.second }?.first
+
+                            if (font != null)
+                            {
+                                Typeface.CustomFallbackBuilder(FontFamily.Builder(font).build())
+                                    .setSystemFallback(fontName).build()
+                            }
+                            else
+                            {
+                                Typeface.create(fontName, Typeface.NORMAL)
+                            }
+                        }
+                        else
+                        {
+                            Typeface.create(fontName, Typeface.NORMAL)
+                        }
+            }
+
+        }
```

Comment:
Far too much nesting!
- Reverse initial 'if' to 'if == then return'
- return at the end of 'if' of mod font
- reverse 'if' that includes the build-version

Alternatively, separate the system font into its own function, like `createSystemFont(fontName:String)`

Reply:
Changed